### PR TITLE
fix: clearFind on individual window close events

### PR DESCRIPTION
**Pull request by ericawright**
Lets try once more

clearFind on individual window close events

Fixes https://github.com/bwinton/TabCenter/issues/1015.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -502,6 +502,14 @@ exports.main = function (options, callbacks) {
   // Override default preferences
   utils.setDefaultPrefs();
 
+  browserWindows.on('close', function (window) {
+    let win = viewFor(window);
+    let mainWindow = win.document.getElementById('main-window');
+    if (mainWindow.getAttribute('toggledon') === 'true') {
```
Comment (bwinton) [inline main.js]: Should we also check for `win.VerticalTabs`?  I mean, I don't think it won't exist ever here, but just in case?

File: main.js [hunk 2]
```diff
@@ -581,7 +589,11 @@ exports.main = function (options, callbacks) {
 
 exports.onUnload = function (reason) {
   for (let window of browserWindows) {
-    viewFor(window).VerticalTabs.clearFind();
+    let win = viewFor(window);
+    let mainWindow = win.document.getElementById('main-window');
+    if (mainWindow.getAttribute('toggledon') === 'true') {
+      win.VerticalTabs.clearFind();
+    }
```
Comment (bwinton) [inline main.js]: I'm starting to think that we should extract this into a function somewhere…  🙂

**Comment by bwinton** [comment]
Looks good to me.  @SoftVision-PaulOiegas, did you want to give it a try and see if it fixes everything before we merge it?  🙂

**Comment by SoftVision-PaulOiegas** [comment]
@ericawright @bwinton The fix looks good, at least for the 4 scenarios mentioned in the bug. One thing I've observed on both Mac OS and Windows but I don't have yet the steps to reproduce. Somehow the number of tabs to be closed, when having multiple windows, is sometimes higher than the real number of tabs. I'll keep an eye on this.

First time I thought the fix was not good, but the branch build is 1.30 and got updated to master 1.31 😄

**Pull request merged**